frontend/src/screens/ShippingAddressScreen.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { Form, Button } from "react-bootstrap";
+import { useDispatch, useSelector } from "react-redux";
+import FormContainer from "../components/FormContainer";
+import { saveShippingAddress } from "../services/cartAction";
+import store from "../store";
+
+const ShippingAddressScreen = ({ history }) => {
+  //
+  // pull put data from Store (state.cart)
+  const cart = useSelector((state) => state.cart);
+  const { shippingAddress } = cart;
+
+  //state for data updated by form onChange actions
+  const [name, setName] = useState(shippingAddress.name);
+  const [address, setAddress] = useState(shippingAddress.address);
+  const [city, setCity] = useState(shippingAddress.city);
+  const [state, setState] = useState(shippingAddress.state);
+  const [zipcode, setZipcode] = useState(shippingAddress.zipcode);
+
+  //
+  const dispatch = useDispatch();
+
+  function validateInput(str) {
+    var validateInput = true;
+    if (/[!@#$%^&*(),.?":{}|<>]/g.test(str) || /\d+/g.test(str)) {
+      validateInput = false;
+    }
+    return validateInput;
+  }
+  const submitHandler = (e) => {
+    e.preventDefault();
+
+    if (validateInput(name) && validateInput(state) && validateInput(city)) {
+      console.log("pass");
+
+      dispatch(saveShippingAddress({ name, address, city, state, zipcode }));
+      console.log(shippingAddress);
+
+      //history.push("/payment");
+    } else {
+      alert("Invalid Input, please check and try again!");
```

Comment:
This error warning is a little vague. I suggest updating it so that it says the inputs do not allow numbers. And ideally specify which field is causing the error.

Reply:
Good advice, I will change the content of Alert to make it more specific.

Reply:
Thanks for updating.